### DISCUSSION TITLE
JSON schema documentation

### DISCUSSION THREAD

**Discussion by zehawki** [vazco/uniforms#916] [Q&A] [answered]
I'm unable to find any documentation on the schemas - esp JSON. How is one supposed to use anything beyond simple text boxes - radio, checkboxes, select (drop down) being the most used.

For select:
I tried using `allowedValues: ['xs', 's', 'm', 'l', 'xl']` in the playground -> doesnt work
I tried using 

```
options: [
        {
          label: 'Developer',
          value: 'developer',
        },
        {
          label: 'Tester',
          value: 'tester',
        }
]
```
also didnt work

![image](https://user-images.githubusercontent.com/8095144/112950572-0dd2cb00-9158-11eb-8c0f-ba86306ec75f.png)

**Comment by radekmie** [comment, marked as answer]
Hi @zehawki. Actually, your code would work totally fine in ajv v6. From v7, all keywords have to be registered, therefore you'd need to add `allowedValues` and/or `options` to the `keywords` in the `Ajv` constructor. Also, uniforms use the `uniforms` keyword, where you can pass all of the props as normal, without registering them. [Here's a playground example](https://uniforms.tools/playground/#?N4IgDgTgpgzlAuIBcICCATd0YwAQAoApAZQHkA5YgYwAsoBbAQwEoQAacCAezBmVEYBXeFxiMAblGQAzRgBs4HISLGSAIlDmMAnsgCMABgMd0ASzEAjOVHQz5ikFoubk8CIKgcwWqlBpc5dCgIOwVPEGhGdFIAOzldJFkwjhhaBkZkEHx8ZlwAXgA-XGAAHRjcXCouGJh4XEYAK3F83BioAHdcVCb8UvKK-rk5AFEIbggYJFw3DzYygdxBOA1ZQTl4Sen3T3mBgGsobXauCHRNgG0Acnk5LnabADV5DxhLtlxLnnhTatf3y8EMVM0hO9FeAF05v0AL7MADcZV2VRqdVSdCYLT6C2-8GsU0uwwAHox6N4oG9dhV4NowFB8VwLA0oFR4BT-hVIDxgt9YFMsQtcKY4kKoKghncbHzpjS6R9ahAhQBzN6DW73dBPOQvKZXQl_OUqy70Q1yQ2E03g3DQqECwXCtqkMDfX5S6m0_HypUqr4_GpSxj41AxGIqiz4gBCDMuVqtNoFgOBoM2_Ntbtll09MWVcdtCZBEDBfMpttwPpduHOxZLxVwTk0-I0klutIgKvEz3TQSbXNbsarJeAtcYzjk-IAKrB4ME2x38VPatO--yS5D-xVrf2N8utwtoABHQSmaDoHUZtxQBCGgBepjAhtqjCnl1XMIRMV20kBLN9lUiU81pjoI-Jz4Gi6S5CmlS_HU7ZyIBwEQC0jTiAAdFUpKmNYoFpEw8KIsu0DwIIEDlPg9BcEEci5IUxT9rB8EiBAZEUZoeHvsuFTAgQ9FAYxKHBOMeAAGRCbgPEIfxYwnDAKHWFm8A0BBa64IRxHlIOQTwIwmGbOJfECdJVpvra0LFtCxlWvhFTIrUuBgUwAG8ScLRUH-UCOQh2Hoiwb67KpJGtB0uAkBQ1DeeGCroIqUBeek7z2YwHmMWxsI5OwIAwP47QAJL2lAozjKEDiMDA2gxFQsTEIIFj0KYiCJPY4QlWVFUxElUgNWE1ogApDAdRl6QxN8VAgNCQA) of these two approaches.
